src/Components/BankDebitForm/BankDebitProvider.tsx

```diff
@@ -133,20 +142,24 @@ export const BankDebitProvider: FC<Props> = ({
 
   return (
     <div data-test="bankTransferSection">
-      <Spacer mt={2} />
-      {clientSecret && (
-        <Elements options={options} stripe={stripePromise}>
-          <BankDebitForm
-            order={order}
-            bankAccountHasInsufficientFunds={bankAccountHasInsufficientFunds}
-            onSetBankAccountHasInsufficientFunds={
-              onSetBankAccountHasInsufficientFunds
-            }
-            onSetIsSavingPayment={onSetIsSavingPayment}
-          />
-        </Elements>
-      )}
-      {bankDebitSetupError && <BankSetupErrorMessage />}
+      <LoadingArea isLoading={isPaymentElementLoading}>
```

Comment:
It seems like this new `isPaymentElementLoading` is one of the most important pieces, but i'm not sure how it works to fix the issue. Is the idea to avoid redundant re-renders using this loading element?

Reply:
`isPaymentElementLoading` is not new, it was already there in `BankDebitForm` and I moved it to `BankDebitProvider`. 

When we used _<PaymentElement />_ component from _@stripe/react-stripe-js_ library in `BankDebitForm`, `isPaymentElementLoading` indicated that `<PaymentElement />` is ready to be used (it has a `onReady` prop that gets fired when the element is ready). We used `<LoadingArea isLoading={isPaymentElementLoading}>` in `BankDebitForm` until `<PaymentElement />`'s `onReady` prop got fired and set `isPaymentElementLoading` to false. This helped us to display a loading spinner to the user until `<PaymentElement />` was ready to be interacted. 

I moved the definition of `isPaymentElementLoading` and the use of `<LoadingArea />` to `BankDebitProvider` in order to account for `commerceCreateBankDebitSetupForOrder` mutation as well. with this change, `isPaymentElementLoading` will remain true until we get the `clientSecret` in `BankDebitProvider` and _<PaymentElement />_  component is ready to be used. 

Overall this change fixes the friction that user would see an empty area until `clientSecret` is received and then see the loading spinner until _<PaymentElement />_  is ready to be used. now they'll just see a loading spinner as soon as they click on 'Add a new account'.